activesupport/lib/active_support/core_ext/enumerable.rb

```diff
@@ -32,34 +32,6 @@ def maximum(key)
     map(&key).max
   end
 
-  # Calculates a sum from the elements.
-  #
-  #  payments.sum { |p| p.price * p.tax_rate }
-  #  payments.sum(&:price)
-  #
-  # The latter is a shortcut for:
-  #
-  #  payments.inject(0) { |sum, p| sum + p.price }
-  #
-  # It can also calculate the sum without the use of a block.
-  #
-  #  [5, 15, 10].sum # => 30
-  #  ['foo', 'bar'].sum # => "foobar"
-  #  [[1, 2], [3, 1, 5]].sum # => [1, 2, 3, 1, 5]
```

Comment:
This example and the one above no longer work. We can't just break this behaviour that was publicly documented; as mentioned in https://github.com/rails/rails/pull/25202#discussion_r65108031, if we're going to do this we'll have to deprecate the current behaviour first.

Personally I don't think it's worth it, since there are plenty of other opportunities to speed up Rails without inconveniencing users by changing public API.

Reply:
this is very strange behavior for a method that calculates the sum of elements 

also, ruby has a method `flatten` for this `[[1, 2], [3, 1, 5]].flatten => [1, 2, 3, 1, 5]`

the good point is that this code will stop working, raises an exception `TypeError (String can't be coerced into Integer)`, and developers will notice it